Makefile

```
@@ -12,7 +12,7 @@ fmt:
 	cargo +nightly fmt
 
 doc:
-	cargo +nightly doc --lib
+	cargo +nightly doc --lib --features=derive,ws,oauth,jsonpatch
```

```suggestion
	RUSTDOCFLAGS="--cfg docsrs" cargo +nightly doc --lib --features=derive,ws,oauth,jsonpatch
```
To see feature tags, if I remember correctly. Appending `-- --cfg docsrs` might work as well.

---

That would be great, but for some reason, neither of those work as expected. At least not locally. I need to explicitly pass the features to get them to build with `cargo +nightly doc --lib`, and `-- --cfg` also doesn't work.

---

I think the `-- --cfg` flag is for `cargo rustdoc` and that only works on a single manifest, whereas `cargo doc` works on a workspace.

`cargo rustdoc` never seems to get the link between `kube` and `kube_derive::CustomResource` (which is re-exported under a feature).

---

Ok `cargo +nightly rustdoc --lib --workspace` picks up on all crates in the workspace correctly, but it still won't grab the default feature list from `kube/Cargo.toml`'s `package.metadata.docs.rs`.

See https://github.com/rust-lang/cargo/issues/8944

---

Ah, I was running the command in `kube/`.

From the root, the following worked for me:

```
RUSTDOCFLAGS="--cfg docsrs" cargo +nightly doc --lib --workspace --features=derive,ws,oauth,jsonpatch --open
```

Documented all crates and got feature tags in `kube`. `--open` opens the documentation in the browser.

---

Ah, `--open` is nice. But was hoping that we could avoid duplicating the docsrs feature list. But whatever, it's only for local dev.